### PR TITLE
feat: add new package components for theme design

### PR DESCRIPTION
**Pull request by Jack-Works**

### CODE REVIEW

**Comment by Jack-Works** [comment]
currently blocked by https://github.com/npm/cli/issues/2291

**Comment by Jack-Works** [comment]
```
npm ERR! notsup Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin"} (current: {"os":"linux","arch":"x64"})
npm ERR! notsup Valid OS:    darwin
npm ERR! notsup Valid Arch:  undefined
npm ERR! notsup Actual OS:   linux
npm ERR! notsup Actual Arch: x64
```

File: .vscode/tasks.recommended.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// This file contains recommended task for this project.
+// You can copy this file as "tasks.json" to make it work.
+{
+    "version": "2.0.0",
+    "tasks": [
+        {
```
Comment (septs) [inline .vscode/tasks.recommended.json:6]: plz update `caveats.md`

Comment (Jack-Works) [reply .vscode/tasks.recommended.json:6]: what?

**Review by septs — APPROVED** [review]

File: .vscode/tasks.recommended.json (hunk 1 above)
Comment (septs) [reply .vscode/tasks.recommended.json:6]: explain why need to run `tsc --watch` and vscode shorthand task.

Comment (Jack-Works) [reply .vscode/tasks.recommended.json:6]: The developer no need to worry about this. Most of the tasks are doing it behind the scene.

Comment (Jack-Works) [reply .vscode/tasks.recommended.json:6]: there are no workflow changes in this PR. When starting webpack, a `tsc -w -b` will be auto started if there is no running instance.

**Pull request merged**